File: 41. First Missing Positive/Solution.java

class Solution {
    public int firstMissingPositive(int[] nums) {
        var numsList = new boolean[nums.length + 1];

        for (var num : nums)  {
            if (num > 0 && num <= nums.length) {
                numsList[num] = true;
            }   
        }

        for (int i = 1; i < numsList.length; i++) {
            if (!numsList[i]) {
                return i;
            }
        }
        return nums.length +1; 
    }
}

File: 41. First Missing Positive/Solution.go
package main



func firstMissingPositive(nums []int) int {
	//hashList := make(map[int]bool )

	hashList := make([]bool, len(nums)+1)
	for _, num := range nums {
		if num > 0 && num <= len(nums) {
			hashList[num] = true
		}
	}

	for i := 1; i <= len(nums); i++ {
		// if _, ok := hashList[i]; !ok {
		// 	return i
		// }
		if ! hashList[i] {
			return i 
		}
	}
	return len(nums) + 1
}